packages/mui-system/test/integration/theme-scoping.test.tsx

```diff
@@ -0,0 +1,192 @@
+  return <div style={{ color: theme.palette.brand }} />;
+}
+
+const joyTheme = joy.extendTheme({
+  components: {
+    JoyButton: {
+      defaultProps: {
+        variant: 'outlined',
+      },
+      styleOverrides: {
+        root: ({ theme }) => ({
+          color: theme.vars.palette.text.primary,
+        }),
+      },
+    },
+  },
+});
+const CustomJoy = joy.styled('div')(({ theme }) => ({
+  fontSize: theme.vars.fontSize.md,
+}));
+
+const materialTheme = material.createTheme({
+  components: {
+    MuiButton: {
+      defaultProps: {
+        variant: 'outlined',
+      },
+      styleOverrides: {
+        root: ({ theme }) => ({
+          color: theme.palette.text.primary,
+        }),
+      },
+    },
+  },
+});
+const CustomMaterial = material.styled('div')(({ theme }) => ({
+  borderRadius: theme.shape.borderRadius,
+}));
+
+const md3Theme = md3.extendTheme({
+  components: {
+    MuiButton: {
+      defaultProps: {
+        variant: 'outlined',
+      },
+      styleOverrides: {
+        root: ({ theme }) => ({
+          color: theme.vars.palette.error.dark,
+        }),
+      },
+    },
+  },
+});
+
+describe('Multiple nested theme providers', () => {
+  const { render } = createRenderer();
+  let originalMatchmedia: any;
+  let storage: Record<string, string> = {};
+  const createMatchMedia = (matches: boolean) => () => ({
+    matches,
+    addListener: () => {},
+    removeListener: () => {},
+  });
+
+  beforeEach(() => {
+    originalMatchmedia = window.matchMedia;
+
+    // Create mocks of localStorage getItem and setItem functions
+    Object.defineProperty(global, 'localStorage', {
+      value: {
+        getItem: spy((key) => storage[key]),
+        setItem: spy((key, value) => {
+          storage[key] = value;
+        }),
+      },
+      configurable: true,
+    });
+
+    // clear the localstorage
+    storage = {};
+    window.matchMedia = createMatchMedia(false) as unknown as typeof window.matchMedia;
+  });
+  afterEach(() => {
+    window.matchMedia = originalMatchmedia;
+  });
+
+  it('[docs] Material UI + Joy UI', () => {
+    const { getByText } = render(
+      <joy.CssVarsProvider theme={{ [joy.THEME_ID]: joyTheme }}>
+        <material.ThemeProvider theme={materialTheme}>
+          <joy.Button
+            sx={(theme) => ({
+              // test `sx`
+              bgcolor: theme.vars.palette.neutral[100],
+            })}
+          >
+            Joy
+          </joy.Button>
+          <material.Button
+            sx={(theme) => ({
+              bgcolor: theme.palette.secondary.light,
+            })}
+          >
+            Material
+          </material.Button>
+        </material.ThemeProvider>
+      </joy.CssVarsProvider>,
+    );
+    // these test if `useThemeProps` works with theme scoping
+    expect(getByText('Joy')).to.have.class(joy.buttonClasses.variantOutlined);
+    expect(getByText('Material')).to.have.class(material.buttonClasses.outlinedPrimary);
```

Comment:
Would be best if we check that the right colors are used here too, for e.g. that styleOverrides work as expected.

Reply:
Added another test.